### PR TITLE
BlobMeta cleanup

### PR DESCRIPTION
**Pull request by millerdev**
The first and last commits are safe now that the new blob db api is used everywhere on all environments. Any envs we do not manage may have errors during deploy if rolling this out at the same time as the form processor blob db api changes that were merged last week (#21631).

Review by commit 🐠 

@esoergel @snopoke

### CODE REVIEW

File: corehq/blobs/s3db.py [hunk 1]
```diff
@@ -291,7 +185,7 @@ def maybe_not_found(throw=None):
         yield
     except ClientError as err:
         if not is_not_found(err):
-            datadog_counter('commcare.blobdb.notfound')
             raise
+        datadog_counter('commcare.blobdb.notfound')
```
Comment (snopoke) [inline corehq/blobs/s3db.py]: how come this moved here?

**Review by snopoke — APPROVED** [review]

**Comment by snopoke** [comment]
build failed:
```
AttributeError: 'RawSQLMigration' object has no attribute 'get_sql'
```

File: corehq/blobs/s3db.py (hunk 1 above)
Comment (millerdev) [reply corehq/blobs/s3db.py]: The previous placement was incorrect. Notice the double-negative `not is_not_found ~> is_found`

File: corehq/sql_proxy_accessors/migrations/0045_form_attachments.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            SQL("DROP FUNCTION IF EXISTS get_form_attachment_by_name(TEXT, TEXT)"),
+            "get_form_attachment_by_name.sql",
+        ),
+        migrator.get_sql(
+            SQL("DROP FUNCTION IF EXISTS get_form_attachments(TEXT)"),
+            "get_form_attachments.sql",
+        ),
+        migrator.get_sql(
+            SQL("DROP FUNCTION IF EXISTS get_multiple_forms_attachments(TEXT[])"),
+            "get_multiple_forms_attachments.sql",
```
Comment (esoergel) [inline corehq/sql_proxy_accessors/migrations/0045_form_attachments.py]: Guessing removing the query files will come in a later PR?

**Review by esoergel — APPROVED** [review]

File: corehq/sql_proxy_accessors/migrations/0045_form_attachments.py (hunk 2 above)
Comment (millerdev) [reply corehq/sql_proxy_accessors/migrations/0045_form_attachments.py]: Technically they should stick around as long as there are any migrations referencing them. In the past we haven't done this very consistently, and it's one of the reasons our migrations break for people who do not keep their database on the cutting edge of master.

**Review by esoergel — APPROVED** [review]

**Pull request merged**